modules/workspace-impl/src/main/java/org/opencastproject/workspace/impl/WorkspaceImpl.java

```diff
@@ -285,6 +298,26 @@ public void activate(ComponentContext cc) {
     staticCollections.add("videosegments");
     staticCollections.add("waveform");
 
+    // Check if we can read from the asset manager locally to avoid downloading files via HTTP
+    String assetManagerDir = null;
+    if (cc != null && cc.getBundleContext() != null) {
+      assetManagerDir = StringUtils.trimToNull(cc.getBundleContext().getProperty(CONFIG_ASSET_MANAGER_ROOT));
+      if (assetManagerDir == null) {
+        assetManagerDir = StringUtils.trimToNull(cc.getBundleContext().getProperty(CONFIG_STORAGE_DIR));
+        if (assetManagerDir != null) {
+          assetManagerDir = new File(assetManagerDir, "archive").getAbsolutePath();
+        }
+      }
+    }
+
+    // Is the asset manager available locally?
+    // Note that this check is not perfect and *might* fail the very first time Opencast is started up since at that
+    // point the asset manager directory does not exist. But nothing breaks if this test fails and this will work
+    // eventually.
+    if (assetManagerDir != null && new File(assetManagerDir).isDirectory()) {
```

Comment:
What about doing this check for each call to read instead?  That way we don't have to worry about startup order.

Reply:
This would mean that you would have an additional io operation for each call. This check may only fail the very first time an Opencast cluster is started. Once it is restarted (something which will inevitably happen at some point, and is likely to happen fast) this speed boost is enabled. Even, if this test fails, it makes things work like they are now. Not slower than that.

That is why I think over the lifespan of an Opencast node, this will be more beneficial.